src/java/org/apache/cassandra/db/filter/ANNOptions.java

```diff
@@ -72,10 +79,11 @@ public void validate(int limit)
      * @param map the map of options in the {@code WITH ANN_OPTION} of a {@code SELECT} query
      * @return the ANN options in the specified {@code SELECT} options, or {@link #NONE} if no options are present
      */
-    public static ANNOptions fromMap(Map<String, String> map)
+    public static ANNOptions fromMap(String keyspace, Map<String, String> map)
     {
+        assert keyspace != null;
         // ensure that all nodes in the cluster are in a version that supports ANN options, including this one
-        Set<InetAddressAndPort> badNodes = MessagingService.instance().endpointsWithVersionBelow(MessagingService.VERSION_DS_11);
+        Set<InetAddressAndPort> badNodes = MessagingService.instance().endpointsWithVersionBelow(keyspace, MessagingService.VERSION_DS_11);
```

Comment:
@adelapena - what do you think about moving this validation to the `validate` method? It seems odd to pass the `keyspace` in to this method since it is only needed for validation and not for building the object from the map. I left it as is for now since it is a minor detail.

Reply:
Makes sense to me, +1